### PR TITLE
fix(jest-dev-server): fix port detection

### LINKED ISSUE #460: jest-dev-server port detection is not working
## 🐛 Bug Report

I think (unless I've misconfigured something) that with the latest version of jest if you have a process running already on the server port then the port detection won't happen, and tests and jest-dev-server will try to continue anyway.

## To Reproduce

1. Start a HTTP server listening on port 3000, for instance with `npx http-server -p 3000`
1. With the test project (or any project using jest 27.4.3 and jest-dev-server) run `npx jest`
1. An error from the server command will be output to the terminal with an EADDRINUSE error

## Expected behavior

In the past jest-dev-server has prompted the user whether they want to kill the existing process.

## Link to repl or repo (highly encouraged)

Repo with minimal example:

https://github.com/lfdebrux/test-jest-dev-server

## Run `npx envinfo --system --binaries --npmPackages expect-puppeteer,jest-dev-server,jest-environment-puppeteer,jest-puppeteer,spawnd --markdown --clipboard`

Paste the results here:

```bash

*** Clipboard option removed - use clipboardy or clipboard-cli directly ***


## System:
 - OS: macOS 11.6.1
 - CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
 - Memory: 86.04 MB / 16.00 GB
 - Shell: 5.1.8 - /usr/local/bin/bash
## Binaries:
 - Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
 - npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
## npmPackages:
 - jest-dev-server: ^6.0.2 => 6.0.2 
```

### PR DESCRIPTION
**Pull request by gregberge**
Fix #460, #281, #233

### CODE REVIEW

**Comment by colinrotherham** [comment]
Hi @gregberge did these changes build correctly into `jest-dev-server@7.0.0`?

I'm still seeing the old `isPortTaken` (vs `isPortBusy`) via the npm code tab:
https://www.npmjs.com/package/jest-dev-server?activeTab=explore

Thanks

**Comment by gregberge** [comment]
@colinrotherham just redeployed, I checked the code and it is correct.

**Comment by colinrotherham** [comment]
@gregberge That's great, thank you. Appreciate it

We had a Dependabot update for `jest-puppeteer@7.0.0` but didn't realise `jest-dev-server` needed a bump too

File: packages/jest-dev-server/src/index.js [hunk 1]
```diff
@@ -91,22 +91,27 @@ async function outOfStin(block) {
...
-      server.once("close", () => resolve(result)).close();
-    });
-  const val = await new Promise((resolve, reject) => {
-    server = net
-      .createServer()
-      .once("error", (err) =>
-        err.code === "EADDRINUSE" ? resolve(cleanupAndReturn(true)) : reject()
-      )
-      .once("listening", () => resolve(cleanupAndReturn(false)))
-      .listen(config.port, config.host);
```
Comment (tmilar) [inline packages/jest-dev-server/src/index.js:107]: Sorry for the comment on this PR which has already been merged! But I think removing `config.host` from here, caused the issue: https://github.com/argos-ci/jest-puppeteer/issues/555. 

I've submitted a new PR addressing this 😁  https://github.com/argos-ci/jest-puppeteer/pull/556

**Pull request merged**